pkg/odo/genericclioptions/runnable.go

```diff
@@ -77,6 +85,16 @@ func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
 		startTelemetry(cmd, err, startTime)
 	}
 	util.LogErrorAndExit(err, "")
+	// TODO: capture ctrl^z
+	captureSignals := []os.Signal{syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT, os.Interrupt, os.Kill}
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, captureSignals...)
```

Comment:
I want to catch the signal so that it can be sent to telemetry, for which I need another notifier, I don't think I can use the StartSignalWatcher one.

Reply:
You are catching the same signal twice, the first time in this line, and the second time in the body of StartSignalWatcher. A proper way would be to add a `signal` parameter to the `handle` function, and to pass the catched signal to the handle, that it could use (instead of getting the value in line 94 from another catch)